x-pack/platform/plugins/shared/stack_connectors/common/auth/schema.ts

```diff
@@ -12,7 +12,12 @@ import { AuthType, SSLCertType } from './constants';
 
 export const authTypeSchema = schema.maybe(
   schema.oneOf(
-    [schema.literal(AuthType.Basic), schema.literal(AuthType.SSL), schema.literal(null)],
+    [
+      schema.literal(AuthType.Basic),
+      schema.literal(AuthType.SSL),
+      schema.literal(AuthType.OAuth2ClientCredentials),
```

Comment:
keeping same schema and schema validation for `webhook` and `cases-webhook` but not adopted by both. `cases-webhook` will throw on config validation instead. This is to avoid duplications or complex conditions inside the schema. This should also make it easier to opt-in oauth2 for `cases-webhook` the day we decide to